docs/api/projections.rst

```diff
@@ -0,0 +1,37 @@
+Projections
+===========
+
+.. currentmodule:: optax.projections
+
+Projections can be used to perform constrained optimization.
+The Euclidean projection onto a set :math:`\mathcal{C}` is:
+
+.. math::
+
+    \text{proj}_{\mathcal{C}}(u) :=
+    \underset{v}{\text{argmin}} ~ ||u - v||^2_2 \textrm{ subject to } v \in \mathcal{C}.
+
+For instance, here is an example how we can project parameters to the non-negative orthant::
+
+    updates, opt_state = optimizer.update(grads, opt_state)
```

Comment:
I suggest to make this a proper doctest (i.e., with lines that start with >>> and code that runs) so that i) users can copy-paste and ii) the code remains tested?

Reply:
Done